extensions/amp-truncate-text/1.0/binary-search.js

```diff
@@ -0,0 +1,107 @@
+/**
```

Comment:
I think this file should be shared with `0.1`. The cross-dep is probably fine.

Reply:
I did that at first, but the cross-dep fails `gulp dep-check` which explicitly disallows deps between versions =/

Reply:
Would it work if we move shared files somewhere into `extensions/amp-truncate-text`, no version dir?

Reply:
If the file can be shared either in the `extensions/amp-truncate` root (no version) or in `src` that would IMO be preferable than the cross-dep allowlist entry.

Reply:
Had to make a couple minor tweaks to get it to work in Preact mode, will highlight them with inline comments to determine if they're safe to share.